File: Seminarii/Seminar 1/src/domain/SortingTask.java
package domain;

import Utils.AbstractSorter;

public class SortingTask extends Task {
    int[] lista;
    AbstractSorter sorter;

    public SortingTask(String id, String desc, int[] lista, AbstractSorter sorter) {
        super(id, desc);
        this.lista = lista;
        this.sorter = sorter;
    }

    @Override
    public void execute() {
        System.out.println("Lista inainte de sortare:");
        for (int i = 0; i < lista.length; i++) {
            System.out.println(lista[i]);
        }
        sorter.sort(lista);
        System.out.println("Lista dupa sortare:");
        for (int i = 0; i < lista.length; i++) {
            System.out.println(lista[i]);
        }
    }
}

File: Laboratoare_teme/Laborator10/Main/Domain/SortingTask.cs
using Main.Factory;

namespace Main.Domain;

public class SortingTask : Task
{
    private readonly AbstractSorter _sorter;
    private readonly int[] _vector;

    public SortingTask(long id, string description, SortingStrategy strategy, int[] vector)
        : base(id, description)
    {
        _vector = vector;
        _sorter = strategy == SortingStrategy.Bubble 
            ? new BubbleSort() 
            : new QuickSort();
    }

    public override void Execute()
    {
        _sorter.Sort(_vector);
        foreach (var i in _vector)
        {
            Console.Write(i + " ");
        }
    }
}